### PR TITLE
Fix restarts on services that use multiple config keys

### PR DESCRIPTION
**Pull request by kargakis**
Fixes https://github.com/openshift/openshift-azure/issues/266

@openshift/sig-azure ptal

### CODE REVIEW

**Comment by kargakis** [comment]
/hold

Thinking about adding a unit test

**Comment by kargakis** [comment]
Hrm, for a `write` unit test I would either have to bump client-go and remove dynamic.ClientPool in favor of dynamic.Interface so I can use the fake implementation (pretty intrusive refactor) or refactor `write` and split out a function that handles the cleanup of the special objects and the deep equality check. Ideally we want to have the former but for now I will resort to the latter and open a follow-up issue for using dynamic.Interface in the future.

**Comment by kargakis** [comment]
Unit tests are done, @openshift/sig-azure ptal

**Comment by kargakis** [comment]
@openshift/sig-azure ptal

File: pkg/addons/client.go [hunk 1]
```diff
@@ -191,22 +191,9 @@ func write(dyn dynamic.ClientPool, grs []*discovery.APIGroupResources, o *unstru
...
+		if !needsUpdate(existing, o) {
 			return
 		}
-		// this part of code should be executed only for updates
-		// TODO: Split flows to separate functions
-		handleSpecialObjects(*existing, *o)
-
-		log.Infof("Update " + KeyFunc(o.GroupVersionKind().GroupKind(), o.GetNamespace(), o.GetName()))
-
-		// TODO: we should have tests that monitor these diffs:
```
Comment (jim-minter) [inline pkg/addons/client.go]: don't think this comment should be deleted (yet)

**Pull request merged**